tendrl/notifier/notification/handlers/snmp_handler.py

```diff
@@ -0,0 +1,251 @@
+
+    def getPDU(self, message):
+        pdu = [
+            ObjectType(ObjectIdentity('1.3.6.1.4.2312.19.1.1'),
+                       OctetString(message))]
+        return pdu
+
+    def trapV2(self, endpoint, message):
+        # Send trap to one endpoint
+        errorIndication, errorStatus, errorIndex, varBinds = next(
+            sendNotification(
+                SnmpEngine(snmpEngineID=OctetString(
+                    hexValue=endpoint.engineid)),
+                CommunityData(endpoint.community, mpModel=1),
+                UdpTransportTarget((endpoint.host, endpoint.port)),
+                ContextData(),
+                'trap',
+                # sequence of custom OID-value pairs
+                self.getPDU(message)))
+        if errorIndication:
+            log(
+                "error",
+                "notifier",
+                {
+                    "message": 'Unable to sent snmp message to %s err:%s %s %s'
+                    % (endpoint.host,
+                       errorIndication,
+                       errorStatus,
+                       errorIndex)
+                }
+            )
+        else:
+            log(
+                "info",
+                "notifier",
+                {
+                    "message": 'Sent snmp message to %s to alert about %s'
+                    % (endpoint.host,
+                       message)
+                }
+            )
+
+    def trapV3(self, endpoint, message):
+        # Send trap to one endpoint
+        errorIndication, errorStatus, errorIndex, varBinds = next(
+            sendNotification(
+                SnmpEngine(snmpEngineID=OctetString(
+                    hexValue=endpoint.engineid)),
+                endpoint.usm_user,
+                UdpTransportTarget((endpoint.host, endpoint.port)),
+                ContextData(),
+                'trap',
+                # sequence of custom OID-value pairs
+                self.getPDU(message)))
+        if errorIndication:
+            log(
+                "error",
+                "notifier",
+                {
+                    "message": 'Unable to sent snmp message to %s err:%s %s %s'
+                    % (endpoint.host,
+                       errorIndication,
+                       errorStatus,
+                       errorIndex)
+                }
+            )
+        else:
+            log(
+                "info",
+                "notifier",
+                {
+                    "message": 'Sent snmp message to %s to alert about %s'
+                    % (endpoint.host,
+                       message)
+                }
+            )
+
+    def send_message(self, endpoint, message):
+        if endpoint.proto == 2:
+            self.trapV2(endpoint, message)
+        elif endpoint.proto == 3:
+            self.trapV3(endpoint, message)
+
+    def dispatch_notification(self, alert):
+        try:
+            self.set_destinations()
+            if (
```

Comment:
Do we need this formatting? It could fit in one line with pep8 compliance:)

Reply:
actually i trimmed  this line but forgot to convert as single line